backend/sublet/serializers.py

```diff
@@ -0,0 +1,72 @@
+from rest_framework import serializers
+
+from sublet.models import Amenity, Favorite, Offer, Sublet, SubletImage
+
+
+class AmenitySerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Amenity
+        fields = "__all__"
+
+
+class FavoriteSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Favorite
+        fields = "__all__"
+
+
+class OfferSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Offer
+        fields = "__all__"
+        read_only_fields = ["id", "created_date", "user"]
+
+    def create(self, validated_data):
+        validated_data["user"] = self.context["request"].user
+        return super().create(validated_data)
+
+
+class SubletSerializer(serializers.ModelSerializer):
+    amenities = AmenitySerializer(many=True, required=False)
+    #favorites = FavoriteSerializer(many=True, required=False, read_only=True)
+    sublettees = OfferSerializer(many=True, required=False, read_only=True)
```

Comment:
I think this will error, same as favorites. Reason why is because `Offers` and `Favorites` are not sub-components of the `Sublet` class, so it is pretty non-trivial to nest these serializers like that. I suggest removing both favorites and offers here and have the frontend just use the `<sublet_id>/offers` and `<sublet_id>/favorites` route to get this info

Reply:
I left amenities in, but removed sublettees here since it's not useful.